src/crate/client/cursor.py

```diff
@@ -53,8 +54,35 @@ def execute(self, sql, parameters=None, bulk_parameters=None):
         self._result = self.connection.client.sql(sql, parameters,
                                                   bulk_parameters)
         if "rows" in self._result:
+            if "col_types" in self._result:
+                col_types = self._result["col_types"]
+                tmp_data = self._result["rows"]
+
+                rows_to_convert = self._get_rows_to_convert_to_date(col_types)
+                tmp_data = self._convert_dates_to_datetime(tmp_data, rows_to_convert)
+
+                self._result["rows"] = tmp_data
+
             self.rows = iter(self._result["rows"])
```

Comment:
Maybe we can come up with a more elegant and efficient implementation. When trying to follow the code, I recognize that it builds up a list of boolean flags which designates which columns to apply the datetime conversion to, right?

Based on that information, the code will then apply the conversion function on each column while mapping each row into memory through the `list()` function.

I am a bit hesitant on this and wanted to share my specific concerns:

a) The code might allocate a significant amount of memory on larger results.
b) On a very large result set, it might even run out of memory.
c) When applying the conversion, there are currently no sanity checks around which test for `NULL`- or other emptyness. I am specifically looking at `float(str(x)[0:10])` here. Also, there is no other exception handling either. We might want to improve on this aspect.

Maybe we can wrap that conversion into a generator function in order to keep the memory footprint low and, while being at it, add some more bells and whistles on the error handling side?